agent/src/main/java/reactor/blockhound/BlockHound.java

```diff
@@ -193,7 +193,27 @@ private BlockHound() {
         }};
 
         private Consumer<BlockingMethod> onBlockingMethod = method -> {
-            throw new Error(String.format("Blocking call! %s", method));
+            Error error = new Error(String.format("Blocking call! %s", method));
+
+            // Strip BlockHound's internal noisy frames from the stacktrace to not mislead the users
+            StackTraceElement[] stackTrace = error.getStackTrace();
+            int length = stackTrace.length;
+            for (int i = 0; i < length; i++) {
+                StackTraceElement stackTraceElement = stackTrace[i];
+                if (!BlockHoundRuntime.class.getName().equals(stackTraceElement.getClassName())) {
+                    continue;
+                }
+
+                if ("checkBlocking".equals(stackTraceElement.getMethodName())) {
+                    if (i + 1 < length) {
+                        System.arraycopy(stackTrace, i + 1, stackTrace, 0, length - i - 1);
```

Comment:
this doesn't change the size of the `stackTrace` array, leading to duplicated stacktrace elements 💥

verified with strings:
```java
String[] array = new String[] { "A", "B", "C", "D" };
System.arraycopy(array, 2, array, 0, 2);

System.out.println(array.length);
for (String s : array) {
	System.out.println(s);
}
```
prints 4 C D C D

you can probably use `Arrays.copyOfRange`, or emulate its core behavior which might perform slightly better (no .class manipulation and no `Array.newInstance`):

```java
System.arraycopy(original, from, copy, 0, Math.min(original.length - from, newLength));
```

Reply:
Wow! Great catch 🤦‍♂ 

Fixing... 😅

Reply:
Fixed. Thanks a lot for spotting it! 👍